s4cmd.py

```diff
@@ -1815,127 +1815,133 @@ def check_dict(self, opt, value):
-      action='store_true', default=False)
-  parser.add_option(
-      '--verbose', help='verbose output', dest='verbose',
-      action='store_true', default=False)
-  parser.add_option(
-      '--debug', help='debug output', dest='debug',
-      action='store_true', default=False)
-  parser.add_option(
-      '--validate', help='(obsolete) validate lookup operation', dest='validate',
-      action='store_true', default=False)
-  parser.add_option(
-      '-D', '--delete-removed',
-      help='delete remote files that do not exist in source after sync',
-      dest='delete_removed', action='store_true', default=False)
-  parser.add_option(
-      '--multipart-split-size',
-      help='size in bytes to split multipart transfers', type=int,
-      default=50 * 1024 * 1024)
-  parser.add_option(
-      '--max-singlepart-download-size',
-      help='files with size (in bytes) greater than this will be downloaded in '
-      'multipart transfers', type=int, default=50 * 1024 * 1024)
-  parser.add_option(
-      '--max-singlepart-upload-size',
-      help='files with size (in bytes) greater than this will be uploaded in '
-      'multipart transfers', type=int, default=4500 * 1024 * 1024)
-  parser.add_option(
-      '--max-singlepart-copy-size',
-      help='files with size (in bytes) greater than this will be copied in '
-      'multipart transfers', type=int, default=100 * 1024 * 1024)
-  parser.add_option(
-      '--batch-delete-size',
-      help='Number of files (<1000) to be combined in batch delete.',
-      type=int, default=1000)
-  parser.add_option(
-      '--last-modified-before',
-      help='Condition on files where their last modified dates are before given parameter.',
-      type='datetime', default=None)
-  parser.add_option(
-      '--last-modified-after',
-      help='Condition on files where their last modified dates are after given parameter.',
-      type='datetime', default=None)
-
-  # Extra S3 API arguments
-  BotoClient.add_options(parser)
-
-  # Combine parameters from environment variable. This is useful for global settings.
-  env_opts = (shlex.split(os.environ[S4CMD_ENV_KEY]) if S4CMD_ENV_KEY in os.environ else [])
-  (opt, args) = parser.parse_args(sys.argv[1:] + env_opts)
-  s4cmd_logging.configure(opt)
-  
-  if opt.version:
-    message('s4cmd version %s' % S4CMD_VERSION)
-  else:
-    # Initalize keys for S3.
-    S3Handler.init_s3_keys(opt)
-    try:
-      CommandHandler(opt).run(args)
-    except InvalidArgument as e:
-      fail('[Invalid Argument] ', exc_info=e)
-    except Failure as e:
-      fail('[Runtime Failure] ', exc_info=e)
-    except BotoClient.NoCredentialsError as e:
-      fail('[Invalid Argument] ', exc_info=e)
-    except BotoClient.BotoError as e:
-      fail('[Boto3Error] %s: %s' % (e.error_code, e.error_message))
-    except Exception as e:
-      fail('[Runtime Exception] ', exc_info=e, stacktrace=True)
+def main():
+  try:
```

Comment:
why is this try block required here?

Reply:
Simply for cleaner exceptions and sticking with current behaviour. Currently only the [Error] message is printed out and process exits. But with this change the exception trace was also being printed. This simply swallows that (though verbose logging will show the exception stacktrace if needed)